tests/tasks/test_blended_skill_talk.py

```diff
@@ -0,0 +1,343 @@
+    >>>    num_examples = sum([len(d['dialog']) // 2 for d in data])
+    >>>    print(f'Number of episodes: {num_episodes:d}')
+    >>>    print(f'Number of examples: {num_examples:d}')
+
+    Output of manual check:
+    Number of episodes: 4819
+    Number of examples: 27018
+    Number of episodes: 1009
+    Number of examples: 5651
+    Number of episodes: 980
+    Number of examples: 5482
+    """
+
+    def test_counts(self):
+
+        with testing_utils.tempdir() as tmpdir:
+            data_path = tmpdir
+
+            opts_episodes_and_examples = [
+                ({'datatype': 'train'}, 4819, 27018),
+                ({'datatype': 'valid'}, 1009, 5651),
+                ({'datatype': 'test'}, 980, 5482),
+            ]
+            for opt, num_episodes, num_examples in opts_episodes_and_examples:
+                full_opt = Opt(
+                    {**opt, 'task': 'blended_skill_talk', 'datapath': data_path}
+                )
+                teacher = create_task_agent_from_taskname(full_opt)[0]
+                self.assertEqual(teacher.num_episodes(), num_episodes)
+                self.assertEqual(teacher.num_examples(), num_examples)
+
+    def test_check_examples(self):
+
+        with testing_utils.tempdir() as tmpdir:
+            data_path = tmpdir
+
+            # Check the first entry (entry_idx==0) of the second episode for the train
+            # set, in order to check the context for an episode that has a WoW topic
+            # string
+            train_opt_and_example = (
+                {'datatype': 'train'},
+                {
+                    'text': "your persona: i just bought a new house with my partner.\nyour persona: i like to make my own coffee.\nLasagne\nOh, I love lasagne. I make my own noodles as well as the sauce. \nWow.  That's amazing.  I read where lasagne originated in Italy during the Middle Ages.  \nOh really!? That is interesting. I am actually italian myself.",
+                    'labels': [
+                        "Awesome. Me and my partner just bought a house. I can't wait to cook in my kitchen."
+                    ],
+                    'context_dataset': 'wizard_of_wikipedia',
+                    'free_turker_message': 'Oh really!? That is interesting. I am actually italian myself.',
+                    'guided_turker_chosen_suggestion': ' ',
+                    'episode_done': False,
+                },
+            )
+            full_opt = Opt(
+                {
+                    **train_opt_and_example[0],
+                    'task': 'blended_skill_talk',
+                    'datapath': data_path,
+                }
+            )
+            teacher = create_task_agent_from_taskname(full_opt)[0]
+            self.assertEqual(
+                teacher.get(episode_idx=1, entry_idx=0), train_opt_and_example[1]
+            )
+
+            # Check the second entry (entry_idx==1) of the second episode for each dataset
+            opts_and_examples = [
```

Comment:
this is quite a huge dump of text. do we really need all of this just to test?

Reply:
Truncated this